benchmark848/benchmark.cpp

```diff
@@ -0,0 +1,348 @@
+};
+/* will be updated later */
+string tableCopyQuery [NUM_TABLES] = {"", "", ""};
+
+const vector<string> tableColumns[NUM_TABLES] = {{"id", "firstName", "lastName", "sex", "email", "phone", "jobTitle"},
+    {"id", "firstName", "lastName", "company", "city", "country", "primaryPhone", "secondaryPhone", "email", "website"},
+    {"id", "name", "website", "country", "description", "foundYear", "industry", "numEmployee"}};
+const int tableNumColumns[3] = {7, 10, 8};
+
+enum AlterType : int
+{
+    DROP_COLUMN = 0,
+    ADD_COLUMN = 1
+};
+
+enum AlterAddColumnType : int
+{
+    INVALID_COLUMN_TYPE = -1,
+    INT64 = 0,
+    BOOL = 1,
+    STRING = 2
+};
+
+enum TestType : int
+{
+    DROP_TABLE = 0,
+    ALTER_TABLE = 1,
+    DELETE_NODE_GROUP = 2
+};
+
+/* Helper functions */
+std::string GenRandomStr(const int len) {
+    static const char alphanum[] =
+        "0123456789"
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+        "abcdefghijklmnopqrstuvwxyz";
+    std::string tmp_s;
+    tmp_s.reserve(len);
+
+    for (int i = 0; i < len; ++i) {
+        tmp_s += alphanum[rand() % (sizeof(alphanum) - 1)];
+    }
+    
+    return tmp_s;
+}
+
+char* GetCmdOption(char ** begin, char ** end, const std::string & option)
+{
+    char ** itr = std::find(begin, end, option);
+    if (itr != end && ++itr != end)
+    {
+        return *itr;
+    }
+    return nullptr;
+}
+
+void UpdateTableCopyQuery(string csvFileDir)
+{
+    tableCopyQuery[0] = "COPY People FROM '" + csvFileDir + +"/people-10000.csv';";
+    tableCopyQuery[1] = "COPY Customer FROM '" + csvFileDir + +"/customers-10000.csv';";
+    tableCopyQuery[2] = "COPY Organization FROM '" + csvFileDir + +"/organizations-10000.csv';";
+}
+
+string RandomSelectTable(string avoidTable)
+{
+    string tableName = avoidTable;
+    while (tableName == avoidTable) {
+        int tableIndex = rand() % NUM_TABLES;
+        tableName = tableNames[tableIndex];
+    }
+
+    return tableName;
+}
+
+int FindTableIndex(string tableName)
+{
+    for (int i = 0; i < NUM_TABLES; i++) {
+        if (tableNames[i] == tableName) {
+            return i;
+        }
+    }
+
+    KU_ASSERT(0);
+    return 0;
+}
+
+void CreateTable(const unique_ptr<Connection> &connection, string tableName)
+{
+    for (auto i = 0; i < NUM_TABLES; i++) {
+        auto name = tableNames[i];
+        if (tableName == name) {
+            connection->query(tableCreateQuery[i]);
+            connection->query(tableCopyQuery[i]);
+        }
+    }
+}
+
+void DropTable(const unique_ptr<Connection> &connection, string tableName)
+{
+    string query = "DROP TABLE " + tableName + ";";
+    connection->query(query);
+}
+
+/* Note that colType is only valid when alterType is ADD_COLUMN. Otherwise, it should be set as INVALID_COLUMN_TYPE */
+void AlterTable(const unique_ptr<Connection> &connection, string tableName, AlterType alterType, string colName, AlterAddColumnType colType)
```

Comment:
Going by the comment, consider specifying a default value for colType (provided the C++ version supports it), so that no value needs to be specified at the caller side if the alter type is not ADD_COLUMN:

```
void alterTable(const unique_ptr<Connection> &connection, string tableName, AlterType alterType, string colName, AlterAddColumnType colType = INVALID_COLUMN_TYPE) {
// existing code
}
```
Now the function may be called as follows:
```
// DROP COLUMN example
 AlterTable(connection, tableName, DROP_COLUMN, dropColName);
// ADD_COLUMN_EXAMPLE
 AlterTable(connection, tableName, ADD_COLUMN, addColName, INT64);
```

Reply:
Fixed in the newest commit